File: Sort/SelectionSort/Ruby/selection_sort_test.rb
require 'minitest/autorun'
require_relative 'selection_sort'


describe "selection_sort" do
  TEST_DATA = [8, 4, 3, 1, 2, 6, 5, 7];

  it "returns the position of the given value in the given list" do
    expect(selection_sort(TEST_DATA)).must_equal([1, 2, 3, 4, 5, 6, 7, 8])
  end

  it "raises NoSuchValueError when the given value dose not exist in the list" do
    expect{ selection_sort([]) }.must_raise(ArrayIsEmptyError)
  end
end

File: Sort/SelectionSort/TypeScript/selectionSort.test.ts
import { selectionSort } from './selectionSort'

describe('SelectionSort', () => {
  it('sorts the array in the right order', () => {
    const data = [8, 4, 3, 1, 2, 6, 5, 7]
    expect(selectionSort(data)).toStrictEqual([1, 2, 3, 4, 5, 6, 7, 8])
  })

  it('raises an error when the given array is empty', () => {
    expect(() => {
      selectionSort([])
    }).toThrow('The given array is empty')
  })
})
